### ISSUE TITLE
Make script templates file a text file.

### ISSUE THREAD

**Issue by JourneyHans** [godotengine/godot-proposals#10960]
### Describe the project you are working on

The tutorial I am studying mentions the feature of script templates, but it uses GDScript while I am using C#. I am confused about how to create the script templates.

### Describe the problem or limitation you are having in your project

Script templates should not be actual script files, they should be text files because they shouldn’t participate in compilation, they are just templates. Additionally, this can lead to naming conflict issues in C#. For example, the two script templates I created below both use the _CLASS_ placeholder, but this clearly causes compilation errors. Therefore, I had to add suffixes to differentiate them. However, this means I need to modify the class name each time I create a script.
![image](https://github.com/user-attachments/assets/c92ace60-cccd-4bd6-8b4d-74c28fbbc686)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Just change the script template files to text files, like other engines or tools do.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

NaN

### If this enhancement will not be used often, can it be worked around with a few lines of script?

1. Template files no longer participate in compilation.
2. Resolve the issue of renaming placeholders.

### Is there a reason why this should be core and not an add-on in the asset library?

As mentioned above.

**Comment by HolonProduction** [comment]
[The docs recommend excluding the folder from the build](https://docs.godotengine.org/en/stable/tutorials/scripting/creating_script_templates.html#default-behaviour-and-overriding-it)  and adding a `.gdignore` file.

Not opposed to scanning for templates with a `.template` suffix as well though.

**Comment by JourneyHans** [comment]
> [The docs recommend excluding the folder from the build](https://docs.godotengine.org/en/stable/tutorials/scripting/creating_script_templates.html#default-behaviour-and-overriding-it) and adding a `.gdignore` file.[文档建议从构建中排除该文件夹](https://docs.godotengine.org/en/stable/tutorials/scripting/creating_script_templates.html#default-behaviour-and-overriding-it)并添加`.gdignore`文件。
> 
> Not opposed to scanning for templates with a `.template` suffix as well though.不过，也不反对扫描带有`.template`后缀的模板。

This is an effective solution that I overlooked. Thank you for the reminder. It indeed addresses the issue of script templates participating in compilation. I believe resolving this issue should take priority over the naming conflict problem.

**Comment by hawkerm** [comment]
Yeah, I feel like these should be ignored by the compiler. I feel like the intent is `_CLASS_` is just the easy way for it to replace the value in the template for whatever you've typed in the dialogue when you create the new class in the editor.
